pkg/controllers/node/health/controller.go

```diff
@@ -155,11 +164,25 @@ func (c *Controller) findUnhealthyConditions(node *corev1.Node) (nc *corev1.Node
 }
 
 func (c *Controller) annotateTerminationGracePeriod(ctx context.Context, nodeClaim *v1.NodeClaim) error {
+	if expirationTimeString, exists := nodeClaim.ObjectMeta.Annotations[v1.NodeClaimTerminationTimestampAnnotationKey]; exists {
+		expirationTime, err := time.Parse(time.RFC3339, expirationTimeString)
```

Comment:
if we can't parse it, should we override it? Or should we fail close?

Reply:
Yeah, it seems reasonable to override it if we can't parse the time.

Reply:
Seems like we fail close then if we can't parse it. Can you make that change? The extra call is definitely warranted here